### PR TITLE
Make keyword argument use possible with ActiveJob

### PR DESCRIPTION
**Pull request by davidenko87**
This PR will enable the use of kwargs inside the of Activejob perform method

### CODE REVIEW

File: spec/sqewer/active_job_spec.rb [hunk 1]
```diff
@@ -92,4 +130,46 @@ class User < ActiveRecord::Base
     
     wait_for { user.reload.active? }.to eq(true)
   end
+
+  it 'Create file by options given with option arguments' do
```
Comment (julik) [inline spec/sqewer/active_job_spec.rb]: Spec titles are usually transformed context-to-example and they tend to read nicer if they conform to `it "does ..."`. "It saves", "it quits", "it removes" and so forth. This is also how most of our repos do it.

File: lib/sqewer/perform_override.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    run_callbacks :perform do
+      args_with_symbolized_options = arguments.map do |a|
+        a.respond_to?(:symbolize_keys) ? a.symbolize_keys : a
+      end
+      perform(*args_with_symbolized_options)
+    end
+  rescue => exception
+    rescue_with_handler(exception) || raise(exception)
+  end
+end
```
Comment (julik) [inline lib/sqewer/perform_override.rb]: Trailing \n

**Comment by julik** [comment]
Closes #11 and seems to be a shortcoming of ActiveJob 4 - it works fine in AJ 5

File: spec/sqewer/active_job_spec.rb [hunk 3]
```diff
@@ -92,4 +130,50 @@ class User < ActiveRecord::Base
     
     wait_for { user.reload.active? }.to eq(true)
   end
-end
+
+  it 'creates a file with option arguments and checks if it exists' do
+    wait_for { @worker.state }.to be_in_state(:running)
+
+    tmpdir = Dir.mktmpdir
```
Comment (linkyndy) [inline spec/sqewer/active_job_spec.rb]: Shouldn't there be a cleanup step for all these temporary directories? If a block is not used, they [are not](http://ruby-doc.org/stdlib-2.0.0/libdoc/tmpdir/rdoc/Dir.html#method-c-mktmpdir) automatically cleaned up.

Comment (davidenko87) [reply spec/sqewer/active_job_spec.rb]: Great suggestion and the Job for deleting files in unused. +1

Comment (dsnipe) [reply spec/sqewer/active_job_spec.rb]: Temp folders usually are cleaned by a system at some point. On linux it is a boot time. OSX probably has the same behaviour.

Comment (linkyndy) [reply spec/sqewer/active_job_spec.rb]: That's true, but I am on the principle that a test should not mutate anything/have side effects besides its context. It's safer and also avoids a lot of potential, hidden issues that may arise.

File: sqewer.gemspec [hunk 4]
```diff
@@ -38,7 +38,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "bundler", "~> 1"
   spec.add_development_dependency "rake", "~> 10.0"
   spec.add_development_dependency "rspec", "~> 3.0"
-  spec.add_development_dependency "activejob", "~> 4.2.6"
+  spec.add_development_dependency "activejob", '~> 4.2.6'
```
Comment (dsnipe) [inline sqewer.gemspec]: Just curiosity – single quotes? :)

File: lib/sqewer/perform_override.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# This module enables the acceptance of kwargs for the perform method inside
+# of ActiveJob Execute. 
+module PerformWithKeywords
+  def perform_now
+    begin
```
Comment (linkyndy) [inline lib/sqewer/perform_override.rb]: I would've left the `def` - `rescue` - `end` on the same indent level. It looks prettier 😊

Comment (julik) [reply lib/sqewer/perform_override.rb]: In this case we can use a naked rescue even.

**Comment by julik** [comment]
Merging since the failures are intermittent and are actually due to the way we test. Good job @davidenko87 👍

**Pull request merged**